### ISSUE TITLE
Ability to determine direction of initial bind from a child to parent

### ISSUE THREAD

**Issue by bblazely** [aurelia/binding#400]
Hi,

I had a brief discussion about this over on Gitter, but couldn't make any headway.

I have a use case where I'd like the default value coming from a child to set the bound value in the parent's view-model initially. Currently it only works from parent to child.

I've created a demo to try and explain what I mean:
https://gist.run/?id=5e68d410d3d5c625b1304b9a3ac8de80

In this demo, the 'child' component has a default value that the parent binds to via a two way bind. Initially though, the 'null' value from the parent is flowing down to the child and overwriting its default value. I can see why this happens, but I'd like to have some control over it...

Ideally, the value from the child, should flow up to the parent instead. I've worked around it by putting in a setTimeout with no delay, which pushes the default set to the next cycle, but that seems a bit clunky.

A one-way bind here also doesn't seem to work as in this case, a one-way denotes information flowing from the parent view-model into the child element's attribute, not the other way around.

Is there a case for being able to control the setting of default values from a child into the parent model, or marking a bind from a child component as read only so that the parent can't affect it?

ie: attribute.one-way-from="localValue" or something? This language wouldn't solve the case of a two way bind where the initial value comes from the child though...

Take for example, a child element that has some sort of 'view state table' that it wants to tell the parent about, but not allow it to directly influence?

TLDR;
For binds between a parent containing a child element:
* Can a two-way bind honour the default value of the child on initial bind?
* Can a one-way bind be considered read-only from the child to the parent?

Thanks,

B.

**Comment by EisenbergEffect** [comment]
This probably isn't going to happen, not merely for technical reasons, but also because data flow from child up to parent is pretty universally considered a bad practice and has been the source of many, many problems in other frameworks. Frankly, I'd be terrified to add something like this to Aurelia. I understand there are rare situations where there may be a legitimate use case, but I'm afraid it's just to dangerous for us to support that as a built-in feature. As you've already discovered, it is possible to make it work. For now, I think your workaround is going to have to be the way to go. Thanks for understanding.

**Comment by bblazely** [comment]
Hi Rob,

No worries, thought there'd be a reason why and I'm more than happy to be told if i'm doing something a little odd. I also suspected it might be a fairly niche requirement...

Thanks for taking the time to explain your thoughts on it.
